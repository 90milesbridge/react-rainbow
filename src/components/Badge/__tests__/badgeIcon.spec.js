import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import BadgeIcon from '../badgeIcon';

describe('<BadgeIcon/> in the Badge component', () => {
    it('should not have children when the isVisible is not passed', () => {
        const component = mount(
            <BadgeIcon />,
        );
        expect(component.children().length).toBe(0);
    });
    it('should have the right class names when the label is not passed', () => {
        const component = renderer.create(
            <BadgeIcon iconName="utility:world" isVisible iconPosition="left" />,
        );
        expect(component).toMatchSnapshot();
    });
    it('should have the right class names when the label is passed and iconPosition is left', () => {
        const component = renderer.create(
            <BadgeIcon label="testing BadgeIcon" isVisible iconName="utility:world" iconPosition="left" />,
        );
        expect(component).toMatchSnapshot();
    });
    it('should have the right class names when the label is passed and iconPosition is right', () => {
        const component = renderer.create(
            <BadgeIcon label="testing BadgeIcon" isVisible iconName="utility:world" iconPosition="right" />,
        );
        expect(component).toMatchSnapshot();
    });
});
