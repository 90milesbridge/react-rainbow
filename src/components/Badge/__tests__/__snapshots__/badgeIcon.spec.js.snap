// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BadgeIcon/> in the Badge component should have the right class names when the label is not passed 1`] = `
<span
  className="slds-icon_container slds-align_absolute-center"
  style={Object {}}
>
  <svg
    aria-hidden={true}
    className="slds-icon slds-icon_xx-small slds-icon-text-default"
    style={Object {}}
  >
    <use
      xlinkHref="/icons/utility-sprite/svg/symbols.svg#world"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    />
  </svg>
</span>
`;

exports[`<BadgeIcon/> in the Badge component should have the right class names when the label is passed and iconPosition is left 1`] = `
<span
  className="slds-icon_container slds-align_absolute-center slds-m-right_xx-small"
  style={Object {}}
>
  <svg
    aria-hidden={true}
    className="slds-icon slds-icon_xx-small slds-icon-text-default"
    style={Object {}}
  >
    <use
      xlinkHref="/icons/utility-sprite/svg/symbols.svg#world"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    />
  </svg>
</span>
`;

exports[`<BadgeIcon/> in the Badge component should have the right class names when the label is passed and iconPosition is right 1`] = `
<span
  className="slds-icon_container slds-align_absolute-center slds-m-left_xx-small"
  style={Object {}}
>
  <svg
    aria-hidden={true}
    className="slds-icon slds-icon_xx-small slds-icon-text-default"
    style={Object {}}
  >
    <use
      xlinkHref="/icons/utility-sprite/svg/symbols.svg#world"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    />
  </svg>
</span>
`;
